core/src/Streamly/Internal/Data/ParserK/Type.hs

```diff
@@ -512,30 +523,34 @@ adaptWith pstep initial extract cont !relPos !usedCount !input = do
         where
 
         go !_ !pst = do
-            pRes <- pstep pst x
-            case pRes of
+            r <- pstep pst x
+            case r of
+                -- Done, call the next continuation
                 ParserD.Done 0 b ->
-                    cont (Success 1 b) (count + 1) (Chunk x)
+                    pure $ Partial 1 (cont (Success 0 b) (count + 1))
```

Comment:
Can having backtrack on `Partial` only do the job?
Ie. Is it possible to remove the backtrack from `Result`

Reply:
That's exactly what we are doing. Result has relative position. When we have backtracked in Partial, now the relative position becomes 0 (`Success 0`).

Reply:
No, I mean. Can we remove the `Int` from `Success`? If it's always 0, we can remove it.